src/game/Creature.cpp

```diff
@@ -2247,6 +2247,27 @@ bool Creature::MeetsSelectAttackingRequirement(Unit* pTarget, SpellEntry const*
         if (selectFlags & SELECT_FLAG_PLAYER && pTarget->GetTypeId() != TYPEID_PLAYER)
             return false;
 
+        if (selectFlags & SELECT_FLAG_PLAYER_CLASS_WARRIOR && (pTarget->GetTypeId() != TYPEID_PLAYER || pTarget->getClass() != CLASS_WARRIOR))
```

Comment:
switch case for select flag, use some arbitrary select flag - number arithmetic, and make all class select flags use same block of code, inside one check for typeid player and in that IF so smth like if(select_flag-number!=getClass) return false;